samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -233,6 +233,30 @@ void handleContainerLaunchFail(String processorId, String containerId, String pr
     }
   }
 
+  /**
+   * Handle the container stop failure for active containers and standby (if enabled).
+   * @param processorId logical id of the container eg 1,2,3
+   * @param containerId last known id of the container deployed
+   * @param containerHost host on which container is requested to be deployed
+   * @param containerAllocator allocator for requesting resources
+   */
+  void handleContainerStopFail(String processorId, String containerId, String containerHost,
+      ContainerAllocator containerAllocator) {
+    if (processorId != null && hasActiveContainerPlacementAction(processorId)) {
+      // Assuming resource acquired on destination host will be relinquished by the containerAllocator,
+      // we mark the placement action as failed, and return.
```

Comment:
In the case of ContainerPlacement actions, request in the allocator is waiting for active container to stop and since the active container stop fails here, you also want to cancel the resource request tied to the placement action

resourceRequestState.cancelResourceRequest(request);

I believe ContainerPlacementMetadata maintains a set the resource-request that we can read, but only need to cancel the latest one from that set (since that is the one that is active)

Reply:
Isnt the container-stop on the active, issued after the resource has already been acquired?
In that case, if the resource has already been acquired, the resource-request for it has already been met, so it should have already been cancelled?